misc/efi-stub/boot.c

```diff
@@ -347,6 +347,51 @@ static inline EFI_STATUS isspace(CHAR8 ch)
     return ((uint8_t)ch <= ' ');
 }
 
+EFI_STATUS reserve_unconfigure_high_memory(void)
+{
+#define PLATFORM_LO_MMIO_SIZE	0x80000000UL
+	UINTN map_size, map_key, desc_size;
+	EFI_MEMORY_DESCRIPTOR *map_buf;
+	UINTN d, map_end;
+	UINTN i;
+	UINT32 desc_version;
+	EFI_STATUS err;
+	UINT64 reserved_hpa;
+	EFI_PHYSICAL_ADDRESS top_addr_space = CONFIG_PLATFORM_RAM_SIZE + PLATFORM_LO_MMIO_SIZE;
+
+	err = memory_map(&map_buf, &map_size, &map_key, &desc_size, &desc_version);
+	if (err != EFI_SUCCESS)
+		goto fail;
+
+	d = (UINTN)map_buf;
+	map_end = (UINTN)map_buf + map_size;
+
+	for (i = 0; d < map_end; d += desc_size, i++) {
+		EFI_MEMORY_DESCRIPTOR *desc;
+		EFI_PHYSICAL_ADDRESS start, end;
+
+		desc = (EFI_MEMORY_DESCRIPTOR *)d;
+		if (desc->Type != EfiConventionalMemory)
+			continue;
+
+		start = desc->PhysicalStart;
+		end = start + (desc->NumberOfPages << EFI_PAGE_SHIFT);
+
+		if (end > top_addr_space) {
+			if (start < top_addr_space)
+				start = top_addr_space;
+			err = emalloc_fixed_addr(&reserved_hpa, end - start, start);
+			if (err != EFI_SUCCESS)
+				break;
```

Comment:
can we print some message here like: "truncate the memory  of region (desc->PhysicalStart, end) to (start, end).

Reply:
Sure. Sorry for forgetting your comments.